lib/fastlane/plugin/ddg_apple_automation/actions/asana_add_comment_action.rb

```diff
@@ -0,0 +1,130 @@
+        end
+
+        task_id = Fastlane::Actions::AsanaExtractTaskIdAction.run(task_url: task_url) if task_url
+
+        if template_name
+          template_content = load_template_file(template_name)
+          return unless template_content
+
+          html_text = process_template_content(template_content)
+          create_story(asana_access_token, task_id, html_text: html_text)
+        else
+          text = "#{comment}\n\nWorkflow URL: #{workflow_url}"
+          create_story(asana_access_token, task_id, text: text)
+        end
+      end
+
+      def self.description
+        "Adds a comment to the Asana task"
+      end
+
+      def self.authors
+        ["DuckDuckGo"]
+      end
+
+      def self.return_value
+        ""
+      end
+
+      def self.details
+        # Optional:
+        ""
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.asana_access_token,
+          FastlaneCore::ConfigItem.new(key: :task_id,
+                                       description: "Asana task ID",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :task_url,
+                                       description: "Asana task URL",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :comment,
+                                       description: "Comment to add to the Asana task",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :template_name,
+                                       description: "Name of a template file (without extension) for the comment. Templates can be found in assets/asana_add_comment/templates subdirectory.
+      The file is processed before being sent to Asana",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :workflow_url,
+                                       description: "Workflow URL to include in the comment",
+                                       optional: true,
+                                       type: String)
+        ]
+      end
+
+      def self.is_supported?(platform)
+        true
+      end
+
+      def self.validate_params(task_id, task_url, comment, template_name, workflow_url)
+        if task_id.nil? && task_url.nil?
+          raise ArgumentError, "Both task_id and task_url cannot be nil. At least one must be provided."
+        end
+
+        if comment.nil? && template_name.nil?
+          raise ArgumentError, "Both comment and template_name cannot be nil. At least one must be provided."
+        end
+
+        if comment && workflow_url.nil?
+          raise ArgumentError, "If comment is provided, workflow_url cannot be nil"
+        end
```

Comment:
We also don't want empty values, so please rewrite all `.nil?` checks in this function as `.to_s.empty?`
(`nil.to_s` equals `""`)